pysages/methods/pbmetad.py

```diff
@@ -0,0 +1,394 @@
+            # CV are same. This need not be the case for PBMetaD as it generates
+            # free energy along each CV separately. We can define separate grids for each CV
+            # but that is not supported yet.
+
+            # Given bins or shape of each CV is same,
+            # we use shape[0] to define the size of grids.
+            grid_potential = np.zeros((shape[0], shape.size), dtype=np.float64)
+            grid_gradient = np.zeros((shape[0], shape.size), dtype=np.float64)
+
+        return MetadynamicsState(
+            bias, xi, heights, centers, sigmas, grid_potential, grid_gradient, 0, 0
+        )
+
+    def update(state, data):
+        # Compute the collective variable and its jacobian
+        xi, Jxi = cv(data)
+
+        # Deposit Gaussian depending on the stride
+        nstep = state.nstep
+        in_deposition_step = (nstep > 0) & (nstep % stride == 0)
+        partial_state = deposit_gaussian(xi, state, in_deposition_step)
+
+        # Evaluate gradient of biasing potential (or generalized force)
+        generalized_force = evaluate_bias_grad(partial_state)
+
+        # Calculate biasing forces
+        bias = -Jxi.T @ generalized_force.flatten()
+        bias = bias.reshape(state.bias.shape)
+
+        return MetadynamicsState(bias, *partial_state[:-1], nstep + 1)
+
+    return snapshot, initialize, generalize(update, helpers, jit_compile=False)
+
+
+def build_gaussian_accumulator(method: ParallelBiasMetadynamics):
+    """
+    Returns a function that given a `MetadynamicsState`, and the value of the CV,
+    stores the next Gaussian that is added to the biasing potential.
+    """
+    periods = get_periods(method.cvs)
+    height_0 = np.array(method.height, dtype=np.float64)
+    T = method.T
+    deltaT = method.deltaT
+    grid = method.grid
+    kB = method.kB
+    beta = 1 / (kB * T)
+    kB_deltaT = kB * deltaT
+
+    if grid is None:
+        evaluate_potential_each_cv = jit(
+            lambda pstate: np.sum(parallelbias_each_cv(*pstate[:4], periods), axis=0)
+        )
+    else:
+        # each index in pstate.grid_idx correpsonds to different CV.
+        # so, we extract it using np.choose. mode='clip' is added for jit compilation.
+        evaluate_potential_each_cv = jit(
+            lambda pstate: np.choose(np.array(pstate.grid_idx), pstate.grid_potential, mode="clip")
+        )
+
+    def next_height(pstate):
+        V = evaluate_potential_each_cv(pstate)
+        w = height_0 * np.exp(-V / kB_deltaT)
+        cv_switching_probability_sum = np.sum(np.exp(-beta * V))
+        cv_switching_probability = np.exp(-beta * V) / cv_switching_probability_sum
+
+        return w * cv_switching_probability
+
+    if grid is None:
+        get_grid_index = jit(lambda arg: None)
+        update_grids = jit(lambda *args: (None, None))
+        should_deposit = jit(lambda pred, _: pred)
+
+    else:
+        grid_mesh = compute_mesh(grid) * (grid.size / 2)
```

Comment:
grid_mesh variable should have the same value as in [metad.py](https://github.com/SSAGESLabs/PySAGES/blob/main/pysages/methods/metad.py) line 236:
`grid_mesh = (compute_mesh(grid) + 1) * (grid.size / 2) + grid.lower`